CONTRIBUTING.md

```diff
@@ -2,18 +2,49 @@
 
 If you're reading this section, you're probably interested in contributing to Starfish.  Welcome and thanks for your interest in contributing!
 
-# Code of conduct
+# How can I contribute?
+Starfish is designed to specify pipeline recipes for image processing. To support this use, the library is composed as a series of `pipeline_component` modules.
+The objects that sub-class `pipeline_component` spawn a command-line interface that should identically track the API of the python library.
 
+A typical starfish run consists of running one or more image processing filter stages, and identifying features through either a spot- or pixel-based approach.
+The identified features are then decoded into the genes that they correspond to by mapping the fluorescence channel (and optionally hybridization round) using a codebook.
+Finally, the filtered data are segmented, identifying which cell each feature belongs to.
+
+Implementing a new module of an existing `pipeline_component` is a simple process. For example, to add another image filtering step, one would:
```

Comment:
I see that the terminology of "module" came from the spec.  I would argue that we should ditch the term "module" as it is ridiculously overloaded.

I would just rephrase this as "Creating a new algorithm for an existing \`pipeline_component\` ..."

Reply:
Also, "For example, to add a new image filter, one would:"